WcaOnRails/app/controllers/payment_controller.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+class PaymentController < ApplicationController
+  def payment_config
+    payment_id = params.require(:payment_id)
+    competition_id = params.require(:competition_id)
+
+    competition = Competition.find(competition_id)
+    stripe_transaction = StripeTransaction.find(payment_id)
+    secret = stripe_transaction.stripe_payment_intent.client_secret
+    render json: { stripe_publishable_key: AppSecrets.STRIPE_PUBLISHABLE_KEY,
+                   connected_account_id: competition.connected_stripe_account_id,
+                   client_secret: secret }
+  end
+
+  def payment_finish
+    attendee_id = params.require(:attendee_id)
+    payment_request = AttendeePaymentRequest.find_by(attendee_id: attendee_id)
+    return redirect_to Microservices::Registrations.competition_register_path(competition, "not_found") unless payment_request.present?
+    competition_id, = payment_request.competition_and_user_id
```

Comment:
This is very tricky code design (and terribly unintuitive by the designers of Ruby IMO!) because one single comma after the variable name makes all the difference.

To avoid confusion, I'd prefer if you created a `competition_id` method in AttendeePaymentRequest and used that here.

Reply:
Looking at the next line below, maybe even make that `def competition` directly in the APR